pkg/network/network.go

```diff
@@ -104,6 +117,20 @@ func (s *SkupperStatus) LinkBelongsToSameSite(linkName string, siteId string, ro
 
 }
 
+func (s *SkupperStatus) GetRouterIndex(site *SiteStatusInfo) (error, int) {
+
+	for index, router := range site.RouterStatus {
+		// Ignore routers that belong to statefulsets for headless services and any other router
+		routerId := strings.Split(router.Router.Name, "/")
+		if strings.HasPrefix(routerId[1], site.Site.Name) {
```

Comment:
[nonblocker]: Accessing a slice index without bounds checking makes me nervous.

Looks like we've been doing things this way for a while now without getting bitten (above and in skupper_kube_network.go as well), and there could be a reason that we _really_ trust the router name data coming from SiteStatusInfo that I am missing.

Reply:
Good point! totally agree, let me know if I have missed any of them. Thanks!

Fixed in f769553f34944b571b0d33582b2a222f811e0bcf